access-token-management/src/AccessTokenManagement/IDPopProofRequestHandler.cs

```diff
@@ -0,0 +1,11 @@
+// Copyright (c) Duende Software. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See LICENSE in the project root for license information.
+
+namespace Duende.AccessTokenManagement;
+
+public interface IDPopProofRequestHandler
+{
+    Task<bool> TryAcquireDPopProof(HttpRequestMessage request, string? dpopNonce, ClientCredentialsToken token, CancellationToken cancellationToken);
+    Task HandleDPopResponse(HttpResponseMessage response, CancellationToken cancellationToken);
```

Comment:
Since this is a designed extensibility point, we should consider if we can add additional context in case that is needed. Maybe the incoming HttpRequest that triggered this? And consider wrapping the parameters in a simple type - this makes it not a signature change to the method if you later need to add more contextual details that would otherwise require a new parameter to the interface (and cause a breaking change).

Reply:
I'll wrap them in a type. But you can always access the incoming request from the response.